File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache=[]
    
    if cacheSize==0:
        return 5*len(cities)
    
    for i in range(len(cities)):
        city = cities[i].lower()
        
        if city in cache:
            cache.remove(city)
            cache.append(city)
            answer+=1
        
        else:
            if len(cache)==cacheSize:
                cache.pop(0)
            cache.append(city)
            answer+=5
        
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        List<String> lst = new ArrayList<>();
        
        if(cacheSize==0){
            return cities.length*5;
        }
        for(String s: cities){
            // 대소구분 x
            String upper = s.toUpperCase();
            if(lst.remove(upper)){
                // cache hit
                answer++;
                lst.add(upper);
            } else {
                // cache miss
                answer+=5;
                if(lst.size()==cacheSize){
                    lst.remove(0);
                }
                lst.add(upper);
            }
        }
        
        return answer;
    }
}